source/SIL.AppBuilder.Portal/src/lib/components/Pagination.svelte

```diff
@@ -0,0 +1,97 @@
+<script lang="ts">
+  export let size: number;
+  export let total: number;
+  export let page: number = 0;
+
+  $: pageCount = Math.ceil(total / size);
+  $: collapse = pageCount > 6;
+  $: hasPreviousPage = page > 0;
+  $: hasNextPage = page < pageCount - 1;
+
+  function index(i: number, page: number): number {
+    if (page <= 3) return i + 2;
+    else if (page > pageCount - 5) return pageCount + i - 5;
+    else return page + i - 1;
+  }
+</script>
+
+<div class="flex flex-row flex-wrap gap-1 w-full max-w-xs md:w-auto md:max-w-none">
+  <span class="input input-bordered flex items-center gap-2 max-w-xs">
+    Show: <!-- TODO: i18n -->
+    <input type="number" name="size" bind:value={size} />
+    / {total}
+  </span>
```

Comment:
I wasn't sure how best to actually do this, so I have refrained from adding anything to the locale JSON until we have discussed this.